modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -7287,10 +7287,18 @@ private InetSocketAddress checkConnection(List<InetSocketAddress> addrs, int tim
                                     sock.connect(addr, perAddrTimeout);
 
                                     liveAddrHolder.compareAndSet(null, addr);
+
+                                    if (log.isInfoEnabled())
+                                        log.info("Successful connection to the server [address=" + addr + "]");
+                                }
+                                else if (log.isInfoEnabled()) {
+                                    log.info("Connection to the server [address=" + addr + "] is ignored. " +
+                                        "Alive address is found at [address=" + liveAddrHolder.get() + "]");
                                 }
                             }
                             catch (Exception ignored) {
-                                // No-op.
+                                U.warn(log, "Failed to connect the socket to the server [address=" + addr +
```

Comment:
If we lose the connection, we log `log.info('Previous node alive status [alive=" + ok' + ...')`. Where `ok` is `false`. At INFO level. We except that connection might be lost. Wy we use WARN here?

Reply:
I guess WARN is redundant. We just want to see the reason, why the connection failed. I switched it to log.info(...)

Reply:
` log.info("Connection check to previous node done: [liveAddr=" + liveAddr
                                        + ", previousNode=" + U.toShortString(previous) + ", addressesToCheck=" +
                                        nodeAddrs + ", connectingNodeId=" + nodeId + ']');` is not enough? Let's fix it here. `checkConnection` is an utility method. It doesn't make any decision. It could contain DEBUGs I think. If we failed to ping previous node, let's say abount it around here.

Reply:
Yes, I agree with you. `checkConnection ` is an utility method. But it turned out, that it's the only place, where we have access to connection results to all node effective addresses. And the main goal for this implementation was to was logs for failed connections, so that we can see the reason why they failed, or why they weren't chosen.